--- fail2ban-0.9.1/config/jail.conf~	2014-10-28 02:49:40.000000000 +0100
+++ fail2ban-0.9.1/config/jail.conf	2015-01-21 11:16:14.946314277 +0100
@@ -33,7 +33,7 @@
 [INCLUDES]
 
 #before = paths-distro.conf
-before = paths-debian.conf
+before = paths-pld.conf
 
 # The DEFAULT allows a global definition of the options. They can be overridden
 # in each jail afterwards.
@@ -723,4 +723,4 @@
 [portsentry]
 enabled  = false
 logpath  = /var/lib/portsentry/portsentry.history
-maxretry = 1
\ Brak znaku nowej linii na końcu pliku
+maxretry = 1
